File: selection_sort.py
def selection_sort(arr):
    n = len(arr)
    for i in range(n - 1):
        print(f"Iterasi {i + 1}:")
        min_idx = i
        # Cari nilai terkecil di sisa array yang belum urut
        for j in range(i + 1, n):
            if arr[j] < arr[min_idx]:
                min_idx = j
                
        # Tukar dengan elemen pertama dari bagian yang belum urut
        arr[i], arr[min_idx] = arr[min_idx], arr[i]
        print(arr)
    return arr

# Data awal
data = [78, 48, 10, 3, 36, 37, 43, 16, 34, 45, 25, 38, 72, 36, 53, 6]

print("=== SELECTION SORT ===")
print(f"Jumlah Data: {len(data)}")
print("Data Awal:", data)
print("-" * 30)

hasil = selection_sort(data.copy())

print("-" * 30)
print("Data Hasil Sorting:", hasil)

File: selection_sort.cpp
#include <iostream>
#include <vector>

using namespace std;

void printArray(const vector<int>& arr) {
    cout << "[";
    for (size_t i = 0; i < arr.size(); i++) {
        cout << arr[i] << (i == arr.size() - 1 ? "" : ", ");
    }
    cout << "]" << endl;
}

void selectionSort(vector<int>& arr) {
    int n = arr.size();
    for (int i = 0; i < n - 1; i++) {
        cout << "Iterasi " << i + 1 << ":" << endl;
        int min_idx = i;
        for (int j = i + 1; j < n; j++) {
            if (arr[j] < arr[min_idx]) {
                min_idx = j;
            }
        }
        swap(arr[i], arr[min_idx]);
        printArray(arr);
    }
}

int main() {
    vector<int> data = {78, 48, 10, 3, 36, 37, 43, 16, 34, 45, 25, 38, 72, 36, 53, 6};
    
    cout << "=== SELECTION SORT ===" << endl;
    cout << "Jumlah Data: " << data.size() << endl;
    cout << "Data Awal: ";
    printArray(data);
    cout << "------------------------------" << endl;
    
    selectionSort(data);
    
    cout << "------------------------------" << endl;
    cout << "Data Hasil Sorting: ";
    printArray(data);
    
    return 0;
}